src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +359,510 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
+                    validationFailed.volumeName = _("Please enter new volume name");
+                }
+                if (poolTypesNotSupportingVolumeCreation.includes(storagePoolType)) {
+                    validationFailed.storagePool = cockpit.format(_("Pool type $0 does not support volume creation"), storagePoolType);
+                }
+                const poolCapacity = parseFloat(convertToUnit(storagePools.find(pool => pool.name == storagePoolName)?.capacity, units.B, diskParams.unit));
+                if (diskParams.size > poolCapacity) {
+                    validationFailed.size = cockpit.format(_("Storage volume size must not exceed the storage pool's capacity ($0 $1)"), poolCapacity.toFixed(2), diskParams.unit);
+                }
+            } else if (mode === USE_EXISTING) {
+                if (mode !== CUSTOM_PATH && !diskParams.existingVolumeName)
+                    validationFailed.existingVolumeName = _("Please choose a volume");
+            }
+
+            return validationFailed;
+        };
+        return validateParams();
+    }, [diskParams, mode, customDiskVerificationFailed, customDiskVerificationMessage, storagePoolName, storagePools]);
+    const validationFailed = validate ? _validationFailed : {};
+
+    let defaultBody;
+    if (dialogLoading) {
+        defaultBody = (
+            <Bullseye>
+                <Spinner />
+            </Bullseye>
+        );
+    } else if (isMediaInsertion) {
+        defaultBody = (
+            <Form onSubmit={e => e.preventDefault()} isHorizontal>
+                <FormGroup fieldId={`${idPrefix}-source`}
+                           id={`${idPrefix}-source-group`}
+                           label={_("Source")} isInline hasNoPaddingTop>
+                    <Radio id={`${idPrefix}-custompath`}
+                           name="source"
+                           label={_("Custom path")}
+                           isChecked={mode === CUSTOM_PATH}
+                           onChange={() => onValueChanged('mode', CUSTOM_PATH)} />
+                    <Radio id={`${idPrefix}-useexisting`}
+                           name="source"
+                           label={_("Use existing")}
+                           isChecked={mode === USE_EXISTING}
+                           onChange={() => onValueChanged('mode', USE_EXISTING)} />
+                </FormGroup>
+                {mode === USE_EXISTING && (
+                    <UseExistingDisk idPrefix={`${idPrefix}-existing`}
+                                     onValueChanged={onValueChanged}
+                                     storagePoolName={storagePoolName}
+                                     existingVolumeName={diskParams.existingVolumeName}
+                                     validationFailed={validationFailed}
+                                     vmStoragePools={storagePools}
+                                     vms={vms}
+                                     vm={vm} />
```

Comment:
These 8 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231114-134752-e3097bd0-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)